### ISSUE TITLE
conflict with the speech history addon

### ISSUE THREAD

**Issue by vortex1024** [mltony/nvda-phonetic-punctuation#7]
when phonnetic punctuation is enabled, speech history doesn't work. I don't know whether one  party holds the exclusive blame, so I'm posting here in hope that the two authors could solve this together. The related speech history issue, where detailed NVDA logs can be found, is:
https://github.com/jscholes/nvda-speech-history/issues/22
thanks

**Comment by mltony** [comment]
Judging by the stacktrace doesn't appear to be a problem with phonetic 
punctuation.
I briefly looked at speech history - it appears that it can't handle 
speech sequences that are already modified by phonetic punctuation. Hard 
to tell the reason without debugging. But from a glance there doesn't 
appear to be a reason to believe that something is wrong with phonetic 
punctuation. Debugging of speech history is needed.

**Comment by vortex1024** [comment]
by speech sequences modified by phonetic punctuation, do you mean sequences with callbacks which call a function to make sound?
i.e. 
sequence=(speech.CallbackCommand(callback=a), text, speech.CallbackCommand(callback=b))
where a and b are functions with play sound?
If not, can you give an example, please? I'm planning to use such a sequence, with phonetic punctuation disabled, to test your theory.

**Comment by mltony** [comment]
If you have some programming skills, here is what you can do.
IN PhoneticPunctuation.py set debug=True, then on line 46 set the log 
file to some real file on your computer and then it will log every 
sequence to that file - logging happens in def preSpeak:
mylog(str(newSequence))
Then you'll have an idea what phonetic punctuation outputs to 
speech.speak and you can further debug why that breaks the other addon.

**Comment by vortex1024** [comment]
Just did that. The interesting thing is that, even with phonetic punctuation disabled with the keyboard shortcut, speech history still doesn’t work. The only way to make it work is if I comment the call to inject speech interceptor in pp.
Does pp modify the speech sequence even when disabled, or could the interception itself be a problem? I think I saw interception chains like this working fine.

**Comment by mltony** [comment]
No, when it's disabled it doesn't modify speech sequence.

Then maybe look at this line:

return originalSpeechSpeak(newSequence, symbolLevel=symbolLevel, *args, 
**kwargs)

Maybe the other addon changed signature of speak() method so that it 
cannot properly process arguments.

**Comment by vortex1024** [comment]
yes, change of signature is certainly the culprit. someone on the speech history issue suggested loading speech history first, thus letting it override speak() before pp, and htis solved the problem. Not the ideal solution, but at least it works

**Comment by tspivey** [comment]
This is a bug in phonetic punctuation.

phonetic punctuation patches both `speech.speak` and `speech.speech.speak`, but only calls the original `speech.speak`. speech history patches `speech.speech.speak`, but not `speech.speak`.

**Comment by XLTechie** [comment]
@mltony, have you considered @tspivey's [comment above](https://github.com/mltony/nvda-phonetic-punctuation/issues/7#issuecomment-1054120293)?

I am concerned about possible interactions between your add-on, Speech History, and my own add-on [Speech Logger](https://addons.nvda-project.org/files/get.php?file=splogger), all of which patch speech.speech.speak.

**Comment by mltony** [comment]
I'll try to find some time to address this in the next release.
